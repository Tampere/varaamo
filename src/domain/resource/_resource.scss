@import './card/resourceCard';
@import './card/slider/resourceCardSlider';
@import './map/resourceMap';
@import './reservationCalendar/resourceReservationCalendar';
