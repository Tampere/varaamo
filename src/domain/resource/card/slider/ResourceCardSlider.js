import * as React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import Button from 'react-bootstrap/lib/Button';

import ResourceCard from '../ResourceCard';

class UnconnectedResourceCardSlider extends React.Component {
  static propTypes = {
    location: PropTypes.object.isRequired,
    history: PropTypes.object.isRequired,
    resources: PropTypes.array.isRequired,
    unit: PropTypes.object.isRequired,
    date: PropTypes.string.isRequired,
    onFavoriteClick: PropTypes.func.isRequired,
    onFilterClick: PropTypes.func.isRequired,
  };

  constructor(props) {
    super(props);
    this.state = {
      activeSlide: 0,
    };
  }

  onNext = () => {
    const { resources } = this.props;
    const { activeSlide } = this.state;
    this.setState({
      activeSlide: Math.min(activeSlide + 1, resources.length - 1),
    });
  };

  onPrevious = () => {
    const { activeSlide } = this.state;
    this.setState({
      activeSlide: Math.max(activeSlide - 1, 0),
    });
  };

  render() {
    const {
      resources,
      location,
      history,
      date,
      unit,
      onFavoriteClick,
      onFilterClick,
    } = this.props;
    const { activeSlide } = this.state;

    return (
      <div className="app-ResourceCardSlider">
        {resources.length > 1 && (
          <Button
            bsStyle="primary"
            className="app-ResourceCardSlider_arrow app-ResourceCardSlider_arrow-left"
            disabled={activeSlide === 0}
            onClick={this.onPrevious}
          />
        )}
        <ResourceCard
          date={date}
          history={history}
          isStacked={resources.length > 1}
          location={location}
          onFavoriteClick={onFavoriteClick}
          onFilterClick={onFilterClick}
          resource={resources[activeSlide]}
          unit={unit}
        />
        {resources.length > 1 && (
          <Button
            bsStyle="primary"
            className="app-ResourceCardSlider_arrow app-ResourceCardSlider_arrow-right"
            disabled={activeSlide === resources.length - 1}
            onClick={this.onNext}
          />
        )}
      </div>
    );
  }
}

export { UnconnectedResourceCardSlider };
export default withRouter(UnconnectedResourceCardSlider);
