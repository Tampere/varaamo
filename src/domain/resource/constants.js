export const resourceProductTypes = Object.freeze({
  RENT: 'rent',
  EXTRA: 'extra',
});

export const resourcePriceTypes = Object.freeze({
  HOURLY: 'hourly',
  DAILY: 'daily',
  WEEKLY: 'weekly',
  FIXED: 'fixed',
});
