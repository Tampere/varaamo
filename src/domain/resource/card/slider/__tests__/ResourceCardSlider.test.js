import React from 'react';
import toJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';

import { UnconnectedResourceCardSlider } from '../ResourceCardSlider';
import unit from '../../../../../common/data/fixtures/unit';
import resource from '../../../../../common/data/fixtures/resource';


describe('ResourceCardSlider', () => {
  test('renders correctly', () => {
    const props = {
      history: {},
      location: {},
      resources: [resource.build()],
      unit: unit.build(),
      date: '30-07-2019',
      onFavoriteClick: jest.fn(),
      onFilterClick: jest.fn(),
    };
    const wrapper = shallow(
      <UnconnectedResourceCardSlider {...props} />,
    );

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
