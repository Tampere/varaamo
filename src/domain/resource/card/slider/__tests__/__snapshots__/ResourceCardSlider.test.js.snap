// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourceCardSlider renders correctly 1`] = `
<div
  className="app-ResourceCardSlider"
>
  <InjectIntl(Connect(InjectT(ResourceCard)))
    date="30-07-2019"
    history={Object {}}
    isStacked={false}
    location={Object {}}
    onFavoriteClick={[MockFunction]}
    onFilterClick={[MockFunction]}
    resource={
      Object {
        "available_hours": Array [],
        "equipment": Array [],
        "id": 1,
        "is_favorite": false,
        "name": Object {
          "en": "Resource (en) - 1",
          "fi": "Resource (fi) - 1",
          "sv": "Resource (sv) - 1",
        },
        "need_manual_confirmation": false,
        "opening_hours": Array [],
        "products": Array [],
        "public": true,
        "required_reservation_extra_fields": Array [],
        "reservable": true,
        "reservable_after": null,
        "slot_size": "00:30:00",
        "supported_reservation_extra_fields": Array [],
        "unit": 1,
        "user_permissions": Object {
          "canMakeReservations": true,
          "isAdmin": false,
        },
      }
    }
    unit={
      Object {
        "id": 1,
        "location": Object {
          "coordinates": Array [
            25.01571,
            60.249935,
          ],
          "type": "Point",
        },
        "name": Object {
          "en": "Unit (en) - 1",
          "fi": "Unit (fi) - 1",
          "sv": "Unit (sv) - 1",
        },
        "opening_hours_today": Object {},
      }
    }
  />
</div>
`;
